File: launchdarkly/csharp/src/OSEG.LaunchDarklyExamples/CopyFeatureFlagExample.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text.Json;

using Org.LaunchDarklyTools.Api;
using Org.LaunchDarklyTools.Client;
using Org.LaunchDarklyTools.Model;

namespace OSEG.LaunchDarklyExamples;

public class CopyFeatureFlagExample
{
    public static void Run()
    {
        var config = new Configuration();
        config.ApiKey = new Dictionary<string, string> {["ApiKey"] = "YOUR_API_KEY"};

        var source = new FlagCopyConfigEnvironment(
            key: "source-env-key-123abc",
            currentVersion: 1
        );

        var target = new FlagCopyConfigEnvironment(
            key: "target-env-key-123abc",
            currentVersion: 1
        );

        var flagCopyConfigPost = new FlagCopyConfigPost(
            comment: "optional comment",
            source: source,
            target: target
        );

        try
        {
            var response = new FeatureFlagsApi(config).CopyFeatureFlag(
                projectKey: null,
                featureFlagKey: null,
                flagCopyConfigPost: flagCopyConfigPost
            );

            Console.WriteLine(response);
        }
        catch (ApiException e)
        {
            Console.WriteLine("Exception when calling FeatureFlagsApi#CopyFeatureFlag: " + e.Message);
            Console.WriteLine("Status Code: " + e.ErrorCode);
            Console.WriteLine(e.StackTrace);
        }
    }
}

File: launchdarkly/typescript-axios/src/CopyFeatureFlagExample.ts
import axios, { AxiosError } from "axios";
import * as api from "launchdarkly_client"

const configuration = new api.Configuration({
  apiKey: "YOUR_API_KEY",
});

const source: api.FlagCopyConfigEnvironment = {
  key: "source-env-key-123abc",
  currentVersion: 1,
};

const target: api.FlagCopyConfigEnvironment = {
  key: "target-env-key-123abc",
  currentVersion: 1,
};

const flagCopyConfigPost: api.FlagCopyConfigPost = {
  comment: "optional comment",
  source: source,
  target: target,
};

new api.FeatureFlagsApi(configuration).copyFeatureFlag(
  "projectKey_string", // projectKey
  "featureFlagKey_string", // featureFlagKey
  flagCopyConfigPost,
).then(response => {
  console.log(response.data);
}).catch((error: Error | AxiosError) => {
  console.log("Exception when calling FeatureFlagsApi#copyFeatureFlag:");
  axios.isAxiosError(error) ? console.log(error.message) : console.log(error);
});
